cypress/e2e/buy-shirt.cy.ts

```diff
@@ -1,22 +1,44 @@
+import {MenuContentPage} from "../page/index";
+import {ProductsListPage} from "../page/index";
+import {ShopingCartPage} from "../page/index";
+import {LoginPage} from "../page/index";
+import {AddressStepPage} from "../page/index";
+import {ShippingStepPage} from "../page/index";
+import {PaymentStepPage} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const productsListPage = new ProductsListPage();
+const shopingCartPage = new ShopingCartPage();
+const loginPage = new LoginPage();
+const addressStepPage = new AddressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepPage();
+
+const email = "aperdomobo@gmail.com";
+const password = "WorkshopProtractor";
+
 describe("Buy a t-shirt", () => {
   it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get("#center_column a.button.ajax_add_to_cart_button.btn.btn-default").click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
-
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-
-    cy.get("button#SubmitLogin.button.btn.btn-default.button-medium").click();
-    cy.get("button.button.btn.btn-default.button-medium").click();
-    cy.get("input#cgv").click();
-    cy.get("button.button.btn.btn-default.standard-checkout.button-medium").click();
-    cy.get("a.bankwire").click();
-    cy.get("button.button.btn.btn-default.button-medium").click();
-
-    cy.get("#center_column > div > p > strong")
-        .should("have.text", "Your order on My Store is complete.");
+    menuContentPage.visitMenuContentPage();
+
+    menuContentPage.goToTShirtMenu();
+
+    productsListPage.addToMyCar();
+
+    productsListPage.proceedCheckout();
+
+    shopingCartPage.goToCheckout();
+
+    loginPage.login(email, password);
+
+    addressStepPage.proceedCheckoutAuth();
+
+    shippingStepPage.checkToS();
+    shippingStepPage.proceedCheckout();
+
+    paymentStepPage.payWithBank();
+    paymentStepPage.proceedCheckout();
+
+    paymentStepPage.verifyOrderComplete();
```

Comment:
in this case I would receive the text you want to verify as parameter, in that way the page object is not attached to an specific message